packages/table/src/useLeafyGreenTable/useLeafyGreenTable.tsx

```diff
@@ -40,24 +45,60 @@ function useLeafyGreenTable<T extends LGRowData, V extends unknown = unknown>({
     () => columnsProp.some(propCol => !!propCol.enableSorting),
     [columnsProp],
   );
+
   const selectColumnConfig = allowSelectAll
     ? baseSelectColumnConfig
     : omit(baseSelectColumnConfig, 'header');
+
   const columns = React.useMemo<Array<LGColumnDef<T, V>>>(
     () => [
       ...(hasSelectableRows ? [selectColumnConfig as LGColumnDef<T, V>] : []),
-      ...columnsProp.map(propColumn => {
-        return {
-          ...propColumn,
-          align: propColumn.align ?? 'left',
-          enableSorting: propColumn.enableSorting ?? false,
-        } as LGColumnDef<T, V>;
-      }),
+      ...columnsProp,
     ],
     [columnsProp, hasSelectableRows, selectColumnConfig],
   );
 
+  /**
+   *  Custom getExpandedRowModel that manipulates rows to include expandedContent.
+   */
+  function getLGExpandedRowModel<TData extends LGTableDataType<T>>() {
+    return (table: Table<TData>) => {
+      const baseExpandedRowModel = getExpandedRowModel<TData>()(table);
+
+      // Return a function that computes the custom RowModel
+      return () => {
+        // Get the default expanded row model by invoking baseExpandedRowModel
+        const rowModel = baseExpandedRowModel();
+        const modifiedRows = [...rowModel.rows];
+
+        for (let i = 0; i < modifiedRows.length; i++) {
+          if (
+            modifiedRows[i].original.renderExpandedContent &&
+            modifiedRows[i].getIsExpanded()
+          ) {
+            const expandedData = {
+              ...modifiedRows[i],
+              id: `${modifiedRows[i].id}-expandedContent`,
+              isExpandedContent: true,
+            };
+            modifiedRows.splice(i + 1, 0, expandedData);
```

Comment:
This could be a pretty expensive thing to do, if this is not `O(1)` - especially for a large dataset, potentially virtualized and with a lot of expanded rows.

See https://stackoverflow.com/questions/5175925/whats-the-time-complexity-of-array-splice-in-google-chrome.